.github/workflows/publish.yaml

```diff
@@ -0,0 +1,490 @@
+          echo '##[set-output name=sha-of-development-branch;]$(echo $(git branch -av | grep remotes/origin/development | grep -oP "development\s+\K\w+"))'
+          echo '##[set-output name=sha-of-current-branch;]$(echo $(git branch -av | grep "*" | grep -oP " \s+\K\w+"))'
+        id: extract_branch
+
+      - name: Display extracted branch information
+        if: runner.os == 'Linux'
+        run: |
+          echo "SHA of current branch: ${{ steps.extract_branch.outputs.sha-of-current-branch }}"
+          echo "SHA of main branch: ${{ steps.extract_branch.outputs.sha-of-main-branch }}"
+          echo "SHA of development branch: ${{ steps.extract_branch.outputs.sha-of-development-branch }}"
+
+      - name: Is this running on development branch?
+        if: runner.os == 'Linux'
+        run: if [ "${{ steps.extract_branch.outputs.sha-of-current-branch}}" = "${{ steps.extract_branch.outputs.sha-of-development-branch}}" ] ; then echo "Running on development branch."; else echo "Not running on development branch."; fi
+
+      - name: Is this running on main branch?
+        if: runner.os == 'Linux'
+        run: if [ "${{ steps.extract_branch.outputs.sha-of-current-branch}}" = "${{ steps.extract_branch.outputs.sha-of-main-branch}}" ] ; then echo "Running on main branch."; else echo "Not running on main branch."; fi
+
+      - name: Fail if attempting to publish from a branch other than main
+        if: runner.os == 'Linux' && github.event.inputs.publish_to_pypi == '1'
+        run: if [ "${{ steps.extract_branch.outputs.sha-of-current-branch}}" = "${{ steps.extract_branch.outputs.sha-of-main-branch}}" ] ; then echo "Running on main branch, can proceed to publish."; else exit 1; fi
+
+      - name: Activate virtual environment (Linux/MacOS)
+        if: runner.os == 'Linux' || runner.os == 'macOS'
+        run: . venv/bin/activate
+
+      - name: Activate virtual environment (Windows)
+        if: runner.os == 'Windows'
+        run: venv\Scripts\Activate.ps1
+
+      - name: Display Pip Version and confirm environment empty
+        run: |
+          pip -V
+          pip freeze
+      - name: Install Dev Dependencies
+        run: pip install -r requirements-dev.txt
+
+      - name: Install Main Package
+        run: pip install -e .
+      - name: Log full installed packages
+        run: pip freeze
+
+      - name: Run pre-commit hooks
+        if: matrix.IS_FASTEST_JOB == true
+        run: |
+          pre-commit install
+          pre-commit run -a
+
+      - name: Run Tests with randomly generated seed
+        if: github.event.inputs.randomlyseed == ''
+        run: pytest -xsvv --cov-report=xml --full-ci --include-slow-tests
+
+      - name: Run Tests with supplied pytest-randomly seed
+        if: github.event.inputs.randomlyseed != ''
+        run: pytest -xsvv --cov-report=xml --full-ci --include-slow-tests --randomly-seed=${{ github.event.inputs.randomlyseed }}
+
+      - name: Confirm Sphinx Docs build correctly
```

Comment:
confirm for this repo whether the sphinx is in `docs/requirements.txt` or `requirements-dev.txt` and make sure if docs to install the dependency. i don't remember in this repo where it is